packages/wrangler/src/assets.ts

```diff
@@ -444,6 +444,16 @@ export function validateAssetsArgsAndConfig(
 		);
 	}
 
+	// Smart placement turned on when using assets
+	if (
+		config?.placement?.mode === "smart" &&
+		config?.assets?.experimental_serve_directly === false
+	) {
+		logger.warn(
+			"Using assets with smart placement enabled may result in poor performance."
```

Comment:
Update verbiage based on @korinne 's feedback, update with doc link

Reply:
`Warning: Turning on Smart Placement in a Worker that is using assets and `serve_directly` set to `false` means that your entire Worker could be moved to run closer to your data source, and all requests will go to that Worker before serving assets. This could result in poor performance as round trip times could increase when serving assets. Read more in the documentation here`